core/trino-spi/src/main/java/io/trino/spi/SplitWeight.java

```diff
@@ -60,29 +65,41 @@ public int hashCode()
     @Override
     public String toString()
     {
-        return Long.toString(value);
+        if (value == UNIT_VALUE) {
+            return "1";
+        }
+        return BigDecimal.valueOf(value, -UNIT_SCALE).stripTrailingZeros().toPlainString();
     }
 
     @JsonCreator
-    public static SplitWeight of(long value)
+    public static SplitWeight fromEncodedValue(long value)
     {
         return value == UNIT_VALUE ? STANDARD_WEIGHT : new SplitWeight(value);
     }
 
+    public static SplitWeight fromProportion(double weight)
```

Comment:
This wouldn't be necessary if we treat weights as "just an integer larger than 0". See other comments.

Reply:
The problem is that the scheduler needs to know how to translate split counts into split weights, such that any connector can participate in the weighting of splits without having to establish a new definition in common. The definition here is that the scheduler has a standard conversion of `1 standard split` to some integer value (and we're trying to keep the exact integer mapping as an internal implementation detail so that we can change the unit value in the future without breaking the SPI consumers). If we say that the consumers declare a floating point value relative to `1.0` as standard- then we can come back later and say `long UNIT_VALUE = 1_000` to change the expressible relative granularity and nothing breaks.